connector_importer_source_sftp/models/source_csv_sftp.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2019 Camptocamp SA (<http://camptocamp.com>)
+# @author: Sebastien Alix <sebastien.alix@camptocamp.com>
+# Copyright 2020 ACSONE SA/NV (<http://acsone.eu>)
+# @author: Simone Orsi <simahawk@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+import logging
+import os
+
+from odoo import fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+# TODO: in the future, split this to a generic mixin not tied to CSV.
+class ImportSourceCSVSFTP(models.Model):
+    """Import source for CSV files on SFTP."""
+
+    _name = "import.source.csv.sftp"
+    _inherit = [
+        "import.source.csv",
```

Comment:
try also inheriting from the mixin again here


```suggestion
        "import.source.csv",
        "import.source.consumer.mixin",
```

Reply:
@ivantodorovich something really unexpected happening here. Seems still not working as expected. I wonder if we could adapt to use `fields.Selection` here?

Reply:
Yes, simplify if needed.

Reply:
Making `source_model` selection attribute as usual list of tuples and adding to the inherited class `selection_add` also doesn't work.

Reply:
@simahawk @sebalix do you have any idea of the issue that `_selection_source_ref_id` override doesn't actually add `new_source`, simply not called.

`connector_importer_source_sftp/models/source_mixin.py`
```
@api.model
def _selection_source_ref_id(self):
    selection = super()._selection_source_ref_id()
    new_source = ("import.source.csv.sftp", "CSV SFTP")
    if new_source not in selection:
        selection.append(new_source)
    return selection
```